resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnSparkHadoopUtil.scala

```diff
@@ -193,8 +193,7 @@ object YarnSparkHadoopUtil {
       sparkConf: SparkConf,
       hadoopConf: Configuration): Set[FileSystem] = {
     val filesystemsToAccess = sparkConf.get(FILESYSTEMS_TO_ACCESS)
-      .map(new Path(_).getFileSystem(hadoopConf))
-      .toSet
+    val isRequestAllDelegationTokens = filesystemsToAccess.isEmpty
```

Comment:
this would mean that if you have your running application accessing different namespaces and you want to add a new namespace to connect to, if you just add the namespace you need the application can break as we are not getting anymore the tokens for the other namespaces.
I'd rather follow @jerryshao's comment about avoiding to crash if the renewal fails, this seems to fix your problem and it doesn't hurt other solutions.

Reply:
The fetch delegation token is inherently heavy. It took `22 seconds` to get 5 tokens:
![image](https://user-images.githubusercontent.com/5399861/42500208-349c797a-8463-11e8-879a-c8b728d20c07.png)

Reply:
Now the `spark.yarn.access.hadoopFileSystems` configuration is invalid, always get all tokens.

Reply:
Users should be able to configure their needed FileSystem for better performance at least.

Reply:
`spark.yarn.access.hadoopFileSystems` is not invalid, it is just needed to access external cluster, which is what it was created for. Moreover, if you use viewfs, the same operations are performed under the hood by Hadoop code. So this seems to be a more general performance/scalability issue on the number of namespaces we support.

Reply:
`spark.yarn.access.hadoopFileSystems` is not used as what you think. I don't think changing the semantics of `spark.yarn.access.hadoopFileSystems` is a correct way.

Basically your problem is that not all the nameservices are accessible in federated HDFS, currently the Hadoop token provider will throw an exception and ignore the following FSs. I think it would be better to try-catch and ignore bad cluster, that would be more meaningful compared to this fix.

If you don't want to get all tokens from all the nameservices, I think you should change the hdfs configuration for Spark. Spark assumes that all the nameservices is accessible. Also token acquisition is happened in application submission, it is not a big problem whether the fetch is slow or not.

Reply:
Then `spark.yarn.access.hadoopFileSystems` only can be used in filesystems that without HA.
Because HA filesystem must configure 2 namenodes info in `hdfs-site.xml`.

Reply:
cc @LantaoJin @suxingfate You should be familiar with this.

Reply:
@wangyum spark.yarn.access.hadoopFileSystems could be set with HA. 
For example:

` --conf spark.yarn.access.namenodes hdfs://cluster1-ha,hdfs://cluster2-ha`
in hdfs-site.xml
`    <property>`
`        <name>dfs.nameservices</name>`
`        <value>cluster1-ha,cluster2-ha</value>`
`    </property>`
`    <property>`
`        <name>dfs.ha.namenodes.cluster1-ha</name>`
`        <value>nn1,nn2</value>`
`    </property>`
`    <property>`
`        <name>dfs.ha.namenodes.cluster2-ha</name>`
`        <value>nn1,nn2</value>`
`    </property>`